### PR TITLE
Make the schedule times changeable

### PR DESCRIPTION
**Pull request by Ana06**

### CODE REVIEW

**Comment by Ana06** [comment]
What happens if an event is already scheduled at 9 and then the conference start hour is changed to 10? I suppose I should delete all EventSchedules that are not in the hours range when updating them.

**Comment by differentreality** [comment]
Isn't the schedule itself that defines when a conference starts?

If there is sth scheduled at 9am, then the conference starts at 9am.

**Comment by Ana06** [comment]
@differentreality No. Currently the public schedules goes from 9:00 to 20:00. We could do what you say in the public schedule. However we need to set start and end hours for the admin schedule, as currently you can not schedule an event at 7:00.

image

That the public schedule itself defines when a conference starts has some things I don't like:
- The hours range in the public schedule will change when an event is scheduled earlier, so maybe it is a little bit confusing for users.
- It is more complicated to implement and it makes almost no difference.

**Comment by differentreality** [comment]
OK so we will enable the organizers to schedule events at any (random) time they want (eg. earlier than 9am). Alright, that would be interesting :)

**Comment by differentreality** [comment]
I am wondering if it makes sense to add times for the Schedule model itself instead of putting more info on Conference.

As far as deleting the earlier talks is concerned, while it makes some sense, what an organizer would probably want to do is move everything 1 hour later instead.

**Comment by Ana06** [comment]
But then we could have schedules with different times, do we want that?

**Comment by differentreality** [comment]
Why not? I mean since we are allowing organizers to make suggestions about how the schedule should look like, that can include options about the starting time of the keynotes. In the end, the selected_schedule will reflect what is decided. 
What do others think?

File: .rubocop.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ Metrics/BlockNesting:
   Max: 4
 
 Metrics/ClassLength:
-  Max: 575
+  Max: 585
```
Comment (differentreality) [inline .rubocop.yml]: Let's disable the cop in conference model all together, it seems to be a regular issue.
https://github.com/openSUSE/osem/pull/1118/files#diff-a8447a840df703618503bdc301e044e9R1
https://github.com/openSUSE/osem/pull/1118/files#diff-a8447a840df703618503bdc301e044e9R1067

**Comment by differentreality** [comment]
Can you please rebase?

**Comment by differentreality** [comment]
This seems to be working :-) Do we want to provision that the hours will not change after a schedule has been selected?

**Comment by Ana06** [comment]
@differentreality 

> Do we want to provision that the hours will not change after a schedule has been selected?

But you could want to make a change after a schedule has been selected right?

**Comment by differentreality** [comment]
Yeah, let's leave this at that. Looks good to be merged to me. @ChrisBr ?

**Pull request merged**